iothub/device/src/Edge/TrustBundleProvider.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client.Extensions;
+using Microsoft.Azure.Devices.Client.HsmAuthentication;
+using Microsoft.Azure.Devices.Client.HsmAuthentication.GeneratedCode;
+using Microsoft.Azure.Devices.Client.TransientFaultHandling;
+using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+
+namespace Microsoft.Azure.Devices.Client.Edge
+{
+    internal class TrustBundleProvider : ITrustBundleProvider
+    {
+        const string DefaultKeyId = "primary";
+
+        static readonly ITransientErrorDetectionStrategy TransientErrorDetectionStrategy = new ErrorDetectionStrategy();
+        static readonly RetryStrategy TransientRetryStrategy =
+            new TransientFaultHandling.ExponentialBackoff(retryCount: 3, minBackoff: TimeSpan.FromSeconds(2), maxBackoff: TimeSpan.FromSeconds(30), deltaBackoff: TimeSpan.FromSeconds(3));
+
+        public async Task SetupTrustBundle(Uri providerUri, string apiVersion, ITransportSettings[] transportSettings)
+        {
+            TrustBundleResponse trustBundleResponse = await this.GetTrustBundleAsync(providerUri, apiVersion).ConfigureAwait(false);
+            var certs = ParseCertificates(trustBundleResponse.Certificate);
+
+            if (certs.Count() == 0)
+            {
+                throw new InvalidOperationException("Trusted certificate is missing.");
```

Comment:
Instead of throwing, can we simply bypass the certificate handling and validation?